src/code/z_message_staff.c

```diff
@@ -17,12 +17,660 @@ void Message_FindCreditsMessage(PlayState* play, u16 textId) {
+                gDPLoadTextureBlock_4b(gfx++, msgCtx->textboxSegment + 0x1900, G_IM_FMT_I, 96, 48, 0,
+                                       G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                                       G_TX_NOLOD, G_TX_NOLOD);
+                gSPTextureRectangle(gfx++, (msgCtx->textPosX + 0x60) << 2, msgCtx->unk12012 << 2,
+                                    (msgCtx->textPosX + 0xC0) << 2, (msgCtx->unk12012 + 0x30) << 2, G_TX_RENDERTILE, 0,
+                                    0, 1 << 10, 1 << 10);
+                gDPPipeSync(gfx++);
+                gDPSetCombineLERP(gfx++, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0, 0, 0, 0, PRIMITIVE, TEXEL0, 0,
+                                  PRIMITIVE, 0);
+                msgCtx->textPosX += 0x20;
+                break;
+
+            case 0x14: // MESSAGE_TEXT_SPEED
+                msgCtx->textDelay = msgCtx->decodedBuffer.schar[++i];
+                break;
+
+            case 0x1A: // MESSAGE_UNSKIPPABLE
+                msgCtx->textUnskippable = true;
+                break;
+
+            case 0x1B: // MESSAGE_TWO_CHOICE
+                msgCtx->textboxEndType = TEXTBOX_ENDTYPE_10;
+                if (msgCtx->msgMode == MSGMODE_TEXT_DISPLAYING) {
+                    msgCtx->choiceTextId = msgCtx->currentTextId;
+                    msgCtx->stateTimer = 4;
+                    msgCtx->choiceIndex = 0;
+                    Font_LoadMessageBoxEndIcon(font, 2);
+                }
+                break;
+
+            case 0x1C: // MESSAGE_THREE_CHOICE
+                msgCtx->textboxEndType = TEXTBOX_ENDTYPE_11;
+                if (msgCtx->msgMode == MSGMODE_TEXT_DISPLAYING) {
+                    msgCtx->choiceTextId = msgCtx->currentTextId;
+                    msgCtx->stateTimer = 4;
+                    msgCtx->choiceIndex = 0;
+                    Font_LoadMessageBoxEndIcon(font, 2);
+                }
+                break;
+
+            case 0x2: // MESSAGE_END
+                if (msgCtx->msgMode == MSGMODE_TEXT_DISPLAYING) {
+                    msgCtx->msgMode = MSGMODE_TEXT_DONE;
+                    if (msgCtx->textboxEndType == TEXTBOX_ENDTYPE_00) {
+                        Audio_PlaySfx(NA_SE_SY_MESSAGE_END);
+                        Font_LoadMessageBoxEndIcon(font, 1);
+                        if (play->csCtx.state == CS_STATE_IDLE) {
+                            func_8011552C(play, DO_ACTION_RETURN);
+                        }
+                    }
+                }
+                *gfxP = gfx;
+                return;
+
+            case 0x10: // MESSAGE_OCARINA
+                if ((i + 1) == msgCtx->textDrawPos) {
+                    Message_HandleOcarina(play);
+                    *gfxP = gfx;
+                    return;
+                }
+                break;
+
+            case 0xE: // MESSAGE_FADE
```

Comment:
I'm curious why the enum is in the comment and the value is still literal, aren't they normally reversed?

Reply:
Yeah, normally the `0xE` wouldn't be there. The main reason is that the `MESSAGE_FADE` probably won't be an enum, but a hex code define. It has to do with text extraction which we haven't done yet. If your curious I would recommend looking at `message_data_fmt.h` in OoT.